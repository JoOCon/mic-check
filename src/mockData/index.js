export const mockUser = {
  userName: 'Tom Hanks',
  password: 'password',
  email: 'THanks@hanks.com',
  location: '20 McAllister St, San Francisco, CA'
};

export const mockUsers = [{
  userName: 'Tom Hanks',
  password: 'password',
  email: 'THanks@hanks.com',
  location: '20 McAllister St, San Francisco, CA'
},
{
  userName: 'Jim Hanks',
  password: 'password',
  email: 'JHanks@hanks.com',
  location: '20 McAllister St, San Francisco, CA'
}];

export const mockItem = {
  name: 'Guitar',
  location: '101 Market St, San Francisco, CA',
  description: 'Its a guitar!',
  image: "https://guitar-image.png",
  id: 1
};

export const mockItem2 = {
  name: 'Amp',
  location: '101 Market St, San Francisco, CA',
  description: 'Its a amp!',
  image: "https://amp-image.png",
  id: 2
};

export const mockItems = [{
  name: 'Guitar',
  location: '101 Market St, San Francisco, CA',
  description: 'Its a guitar!',
  image: "https://guitar-image.png",
  id: 1
}, 
{
  name: 'Amp',
  location: '101 Market St, San Francisco, CA',
  description: 'Its a amp!',
  image: "https://amp-image.png",
  id: 2
}];

export const mockSenderLocation = '101 Market St, San Francisco, CA';
export const mockDeliveryLocation = '20 McAllister St, San Francisco, CA';

export const mockQuoteData = {
  kind: "delivery_quote",
  fee: 1140,
  created: "2018-09-13T00:40:02Z",
  pickup_duration: 5,
  expires: "2018-09-13T00:45:01Z",
  currency: "usd",
  currency_type: "USD",
  duration: 45,
  dropoff_eta: "2018-09-13T01:25:02Z",
  id: "dqt_Ltalsge5ktl1YV"
};

export const mockOrderInfo = {
  manifest: 'a box of shoes',
  'manifest_items[].name': 'Nikes',
  'manifest_items[].size': '2',
  pickup_address: '101 Market St, San Francisco, CA',
  pickup_name: 'Allen',
  pickup_phone_number: 3033334444,
  dropoff_address: '20 McAllister St, San Francisco, CA',
  dropoff_name: 'Tim',
  dropoff_phone_number: 2223334444,
  dropoff_notes: 'Tim@timmmy.com'
};