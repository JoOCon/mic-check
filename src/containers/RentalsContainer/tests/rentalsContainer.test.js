import React from 'react';
import { shallow } from 'enzyme';
import { RentalsContainer, mapStateToProps } from '../';
import * as mockData from '../../../mockData';

describe('RentalsContainer tests', () => {
  let wrapper;
  const { mockItems } = mockData;
  beforeEach(() => {
    wrapper = shallow( 
      <RentalsContainer 
        rentals={mockItems}
      /> 
    );
  });

  it('renders the RentalsContainer with the correct props', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('MapStateToProps', () => {
    it('should have a rentals data object in props', () => {
      const mockState = {rentals: mockItems};
      const expected = {rentals: mockItems};
      const props = mapStateToProps(mockState);
      expect(props).toEqual(expected);
    });
  });
});
