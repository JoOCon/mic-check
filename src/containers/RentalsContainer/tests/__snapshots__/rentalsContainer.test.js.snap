// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RentalsContainer tests renders the RentalsContainer with the correct props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RentalsContainer
    rentals={
        Array [
            Object {
              "description": "Its a guitar!",
              "id": 1,
              "image": "https://guitar-image.png",
              "location": "101 Market St, San Francisco, CA",
              "name": "Guitar",
              "rentalPrice": 50,
            },
            Object {
              "description": "Its a amp!",
              "id": 2,
              "image": "https://amp-image.png",
              "location": "101 Market St, San Francisco, CA",
              "name": "Amp",
              "rentalPrice": 50,
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(RentalCard)
          rental={
                    Object {
                              "description": "Its a guitar!",
                              "id": 1,
                              "image": "https://guitar-image.png",
                              "location": "101 Market St, San Francisco, CA",
                              "name": "Guitar",
                              "rentalPrice": 50,
                            }
          }
/>,
        <Connect(RentalCard)
          rental={
                    Object {
                              "description": "Its a amp!",
                              "id": 2,
                              "image": "https://amp-image.png",
                              "location": "101 Market St, San Francisco, CA",
                              "name": "Amp",
                              "rentalPrice": 50,
                            }
          }
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "43143b80-bb74-11e8-a00f-3f159937b244",
        "nodeType": "class",
        "props": Object {
          "rental": Object {
            "description": "Its a guitar!",
            "id": 1,
            "image": "https://guitar-image.png",
            "location": "101 Market St, San Francisco, CA",
            "name": "Guitar",
            "rentalPrice": 50,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "43146290-bb74-11e8-a00f-3f159937b244",
        "nodeType": "class",
        "props": Object {
          "rental": Object {
            "description": "Its a amp!",
            "id": 2,
            "image": "https://amp-image.png",
            "location": "101 Market St, San Francisco, CA",
            "name": "Amp",
            "rentalPrice": 50,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(RentalCard)
            rental={
                        Object {
                                    "description": "Its a guitar!",
                                    "id": 1,
                                    "image": "https://guitar-image.png",
                                    "location": "101 Market St, San Francisco, CA",
                                    "name": "Guitar",
                                    "rentalPrice": 50,
                                  }
            }
/>,
          <Connect(RentalCard)
            rental={
                        Object {
                                    "description": "Its a amp!",
                                    "id": 2,
                                    "image": "https://amp-image.png",
                                    "location": "101 Market St, San Francisco, CA",
                                    "name": "Amp",
                                    "rentalPrice": 50,
                                  }
            }
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "43143b80-bb74-11e8-a00f-3f159937b244",
          "nodeType": "class",
          "props": Object {
            "rental": Object {
              "description": "Its a guitar!",
              "id": 1,
              "image": "https://guitar-image.png",
              "location": "101 Market St, San Francisco, CA",
              "name": "Guitar",
              "rentalPrice": 50,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "43146290-bb74-11e8-a00f-3f159937b244",
          "nodeType": "class",
          "props": Object {
            "rental": Object {
              "description": "Its a amp!",
              "id": 2,
              "image": "https://amp-image.png",
              "location": "101 Market St, San Francisco, CA",
              "name": "Amp",
              "rentalPrice": 50,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;
